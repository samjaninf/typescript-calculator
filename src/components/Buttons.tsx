import { BoundActions } from '../types'
import { StyleAttribute, css, hover } from 'glamor'
import { h } from 'preact'
import ButtonGroup from './ButtonGroup'
import OperationButton from './OperationButton'
import InputButton from './InputButton'

type ButtonsProps = {
  actions: BoundActions,
}

const greenButton: StyleAttribute = css(
  { background: '#3bf3a9' },
  hover({ color: '#3bf3a9' }),
)
const zeroButton: StyleAttribute = css({
  flex: '50%',
  paddingLeft: '.9em',
  textAlign: 'left',
})

const Buttons = ({
  actions: {
    add,
    clear,
    divide,
    equals,
    input,
    multiply,
    percent,
    posneg,
    subtract,
  },
}: ButtonsProps): JSX.Element => (
  <div>
    <ButtonGroup>
      <OperationButton action={clear} className={greenButton} text="C" />
      <OperationButton action={posneg} className={greenButton} text="+/&minus;" />
      <OperationButton action={percent} className={greenButton} text="%" />
      <OperationButton action={divide} text="&divide;" />
    </ButtonGroup>
    <ButtonGroup>
      <InputButton action={input} value="7" />
      <InputButton action={input} value="8" />
      <InputButton action={input} value="9" />
      <OperationButton action={multiply} text="&times;" />
    </ButtonGroup>
    <ButtonGroup>
      <InputButton action={input} value="4" />
      <InputButton action={input} value="5" />
      <InputButton action={input} value="6" />
      <OperationButton action={subtract} text="&minus;" />
    </ButtonGroup>
    <ButtonGroup>
      <InputButton action={input} value="1" />
      <InputButton action={input} value="2" />
      <InputButton action={input} value="3" />
      <OperationButton action={add} text="+" />
    </ButtonGroup>
    <ButtonGroup>
      <InputButton action={input} className={zeroButton} value="0" />
      <InputButton action={input} value="." />
      <OperationButton action={equals} text="=" />
    </ButtonGroup>
  </div>
)

export default Buttons
