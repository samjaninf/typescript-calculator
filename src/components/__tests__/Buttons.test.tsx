import Buttons from '../Buttons'
import snapshot from './utils/snapshot'

const actions = { add: () => ({ type: 'ADD' }) }

describe('Buttons', () => {
  it('renders', () => snapshot(Buttons, { actions }))
})
