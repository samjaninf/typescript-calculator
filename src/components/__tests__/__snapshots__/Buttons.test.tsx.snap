// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders 1`] = `
"<div>
  <ButtonGroup>
    <OperationButton class text=\\"C\\"></OperationButton>
    <OperationButton class text=\\"+/−\\"></OperationButton>
    <OperationButton class text=\\"%\\"></OperationButton>
    <OperationButton text=\\"÷\\"></OperationButton>
  </ButtonGroup>
  <ButtonGroup>
    <InputButton value=\\"7\\"></InputButton>
    <InputButton value=\\"8\\"></InputButton>
    <InputButton value=\\"9\\"></InputButton>
    <OperationButton text=\\"×\\"></OperationButton>
  </ButtonGroup>
  <ButtonGroup>
    <InputButton value=\\"4\\"></InputButton>
    <InputButton value=\\"5\\"></InputButton>
    <InputButton value=\\"6\\"></InputButton>
    <OperationButton text=\\"−\\"></OperationButton>
  </ButtonGroup>
  <ButtonGroup>
    <InputButton value=\\"1\\"></InputButton>
    <InputButton value=\\"2\\"></InputButton>
    <InputButton value=\\"3\\"></InputButton>
    <OperationButton text=\\"+\\"></OperationButton>
  </ButtonGroup>
  <ButtonGroup>
    <InputButton class value=\\"0\\"></InputButton>
    <InputButton value=\\".\\"></InputButton>
    <OperationButton text=\\"=\\"></OperationButton>
  </ButtonGroup>
</div>"
`;
